server/proto/merger_test.go

```diff
@@ -0,0 +1,143 @@
+package proto_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	stencilProto "github.com/odpf/stencil/server/proto"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func TestMerge(t *testing.T) {
+
+	t.Run("Run all cases", func(t *testing.T) {
+		for _, test := range []struct {
+			number      int
+			description string
+		}{
+			{1, "should able to merge on simple new fields"},
+			{2, "should able to merge on simple rename of fields"},
+			{3, "should able to merge new field in nested message"},
+			{4, "should able to merge on new field of nested message type"},
+			{5, "should able to merge on new field of message type"},
+			{6, "should able to merge on new field with dependency on well known types"},
+			{7, "should able to merge on new field with dependency on other file"},
+			{8, "should able to merge on addition of package name, options, imports"},
+			{9, "should able to merge on deletion of message without options specified"},
+			{10, "should able to merge on deep nested message"},
+			{11, "should be able to merge on message with options in field"},
+			{12, "should able to merge on deletion of message with options specified"},
+		} {
+			t.Run(test.description, func(t *testing.T) {
+				runTest(t, test.number)
+			})
+		}
+	})
+}
+
+func runTest(t *testing.T, testNumber int) {
+	previous, current, expected := getTestData(t, testNumber)
+	expectedFDS := &descriptorpb.FileDescriptorSet{}
+	err := proto.Unmarshal(expected, expectedFDS)
+	assert.Nil(t, err)
+	expectedFiles, err := protodesc.NewFiles(expectedFDS)
+	assert.Nil(t, err)
+	actual, err := stencilProto.Merge(current, previous)
```

Comment:
we can do byte comparison along with existing assertions. if we enable `Deterministic ` flag in [MarshalOptions](https://pkg.go.dev/google.golang.org/protobuf@v1.27.1/proto#MarshalOptions).